scripts/s3.py

```diff
@@ -0,0 +1,295 @@
+    # Setup logging.
+    logger = logging.getLogger(__name__)
+
+    # Find CloudFront id for the given bucket name.
+    cloudfront_id = CLOUDFRONT_ID_BY_BUCKET.get(bucket_name)
+    if cloudfront_id is None:
+        logger.warning("Could not find a CloudFront id for the S3 bucket '%s'" % bucket_name)
+        return
+
+    logger.info("Creating invalidation for %i keys in the CloudFront distribution '%s'" % (len(keys), cloudfront_id))
+
+    # Connect to CloudFront.
+    cloudfront = boto3.client("cloudfront")
+
+    # Create the invalidation batch. Top-level keys require a "/" prefix for
+    # proper invalidation.
+    invalidation_batch = {
+        "Paths": {
+            "Quantity": len(keys),
+            "Items": ["/%s" % key for key in keys]
+        },
+        "CallerReference": str(time.time())
+    }
+    logger.debug("Invalidation batch: %s" % str(invalidation_batch))
+
+    # Create the invalidation.
+    response = cloudfront.create_invalidation(
+        DistributionId=cloudfront_id,
+        InvalidationBatch=invalidation_batch
+    )
+    logger.debug("CloudFront response: %s" % str(response))
+
+    return response
+
+
+def push(bucket_name, files, dryrun=False):
+    """Push the given files to the given S3 bucket and optionally invalidate the
+    cache for a given CloudFront id.
+
+    Args:
+        bucket_name: S3 bucket to pull from
+        files: a list of local files to upload to the given bucket
+        dryrun: boolean indicating whether files should be downloaded or not
+    """
+    # Setup logging.
+    logger = logging.getLogger(__name__)
+
+    # Create a distinct list of files to push.
+    files = list(set(files))
+
+    # Confirm that all given file paths are proper files.
+    for file_name in files:
+        assert os.path.isfile(file_name), "The requested input '%s' is not a proper file" % file_name
+
+    # Connect to S3.
+    s3 = boto3.resource("s3")
+    bucket = s3.Bucket(bucket_name)
+
+    # Upload local files, stripping directory names from the given file paths
+    # for the S3 keys.
+    logger.info("Uploading %i files to bucket '%s'" % (len(files), bucket_name))
+    s3_keys = []
+    for file_name in files:
+        s3_key = os.path.split(file_name)[-1]
+        s3_keys.append(s3_key)
+        logger.debug("Uploading '%s' as '%s'" % (file_name, s3_key))
+
+        if not dryrun:
+            # Open uncompressed file to be uploaded.
+            with open(file_name, "rb") as fh:
+                # Compress the file in memory.
+                compressed_fh = io.BytesIO()
+                with gzip.GzipFile(fileobj=compressed_fh, mode="wb") as gz:
+                    shutil.copyfileobj(fh, gz)
+
+                compressed_fh.seek(0)
+
+                # Upload the compressed file with associated metadata.
+                bucket.upload_fileobj(
+                    compressed_fh,
+                    s3_key,
+                    {"ContentEncoding": "gzip", "ContentType": "application/json"}
```

Comment:
I think this is the cause of the errors `WARNING:__main__:Skipping unsupported file type for file 'img_zika.png'`

While compression should (probably) be enforced, I think we should "guess" the contentType here.